Events.php

```diff
@@ -69,6 +69,9 @@ public static function onRestApiAddRules()
 
             //space membership
             ['pattern' => 'space/<spaceId:\d+>/membership/<userId:\d+>', 'route' => 'smartVillage/space/membership/create', 'verb' => 'POST'],
+
+            //E-mail and Web Notification Setting
+            ['pattern' => 'settings', 'route' => 'smartVillage/setting/notification/save-settings', 'verb' => 'PUT'],
```

Comment:
with this event some variations of saving notification settings are possible right? do we always need to send all options or is it possible to send one setting and keep others untouched?

furthermore we need an endpoint to get all settings please.

Reply:
Hi @donni106 
Yes we can update only one setting using this endpoint (see the attached screenshot)

![image](https://user-images.githubusercontent.com/56824739/180712465-cf8cf44c-209a-4cd7-b134-4ed97a205f59.png)

**Note: we have to pass both email and web key for a setting otherwise it will throw an error (example)**

![image](https://user-images.githubusercontent.com/56824739/180712649-b68c11e8-b4e9-474d-abfa-62d6f85f2968.png)

### Update all setting

![image](https://user-images.githubusercontent.com/56824739/180712983-64889442-8eaf-4f19-8cf4-e34764127ef0.png)

Note: If the value is disabled (can't change) and you want to update that value, in that case, it will throw an error (see the screenshot below) 

![image](https://user-images.githubusercontent.com/56824739/180713174-4079d6d9-eb16-4c28-a750-83a7fbda7f20.png)

**So, it is better to remove that setting from body**

Reply:
> furthermore we need an endpoint to get all settings please.

Okay, we will make a new endpoint /setting/notification to see all the setting status in the same branch (issue-9) right?

Reply:
Hi @donni106 
Yes we can update only one setting using this endpoint (see the attached screenshot)
### Update only one setting

![image](https://user-images.githubusercontent.com/56824739/180712465-cf8cf44c-209a-4cd7-b134-4ed97a205f59.png)

**Note: we have to pass both email and web key for a setting otherwise it will throw an error (example)**

![image](https://user-images.githubusercontent.com/56824739/180712649-b68c11e8-b4e9-474d-abfa-62d6f85f2968.png)

### Update all setting

![image](https://user-images.githubusercontent.com/56824739/180712983-64889442-8eaf-4f19-8cf4-e34764127ef0.png)

Note: If the value is disabled (can't change) and you want to update that value, in that case, it will throw an error (see the screenshot below) 

![image](https://user-images.githubusercontent.com/56824739/180713174-4079d6d9-eb16-4c28-a750-83a7fbda7f20.png)

**So, it is better to remove that setting from the body**

Reply:
> furthermore we need an endpoint to get all settings please.

Okay, we will make a new endpoint /setting/notification to see all the setting status in the same branch (issue-9).

Reply:
thanks. i saw a problem with merge commit and updated the branch. now it is up to date with latest `main` + changes you did in the one commit for adding the endpoint for settings. now you can work on top of this to add the getter.

Reply:
Hi @donni106 
We facing the problem while rebasing issue-9 branch with main. so can you suggest to us what approach should we take to make commit cleaner without merging issues with the main

See the attached screenshot)

![image](https://user-images.githubusercontent.com/103625795/180749249-3c6abab0-8f4b-459d-844f-320e34272e8a.png)

Reply:
i don't think that you need to rebase. the branch on github is cleaned up with my cleanup earlier. 
i always use visual tool to handle git more structured and clearly.

<img width="585" alt="Bildschirmfoto 2022-07-25 um 12 01 23" src="https://user-images.githubusercontent.com/1942953/180751504-9c08ba1d-53e9-4ade-b694-03d9b28d32e5.png">

you should be able to reset you local branch to the origin one .

Reply:
Okay @donni106 
Thanks!